File: src/main/java/com/gicm/student_management_system/entity/AdditionalStudentInfo.java
package com.gicm.student_management_system.entity;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "additional_student_info")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdditionalStudentInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Extra fields for a student, split out from {@link Student}.
     */
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "common_student_id", nullable = false, unique = true)
    private Student commonStudent;

    // --- Japanese-school / international related ---
    @Column(name = "name_in_japanese")
    private String nameInJapanese;

    @Column(name = "passport_number")
    private String passportNumber;

    @Column(name = "current_japan_level")
    private String currentJapanLevel;

    @Column(name = "japan_travel_experience")
    private Boolean japanTravelExperience;

    @Column(name = "coe_application_experience")
    private Boolean coeApplicationExperience;

    @Column(name = "passed_highest_jlpt_level")
    private String passedHighestJlptLevel;

    // --- Contact / family (not in CommonStudent lines 29-85) ---
    @Column(name = "secondary_phone")
    private String secondaryPhone;

    @Column(name = "father_name")
    private String fatherName;

    // --- Preferences / lifestyle / admin ---
    @Column(name = "desired_job_type")
    private String desiredJobType;

    @Column(name = "other_desired_job_type")
    private String otherDesiredJobType;

    @Column(name = "is_smoking")
    private Boolean isSmoking;

    @Column(name = "is_alcohol_drink")
    private Boolean isAlcoholDrink;

    @Column(name = "have_tatto")
    private Boolean haveTatto;

    @Column(name = "hostel_preference")
    private Boolean hostelPreference;

    @Column(name = "memo_notes", length = 2000)
    private String memoNotes;

    @Column(name = "attending_class_related_status")
    private String attendingClassRelatedStatus;

    @Column(name = "contact_viber")
    private String contactViber;

    @Column(name = "schedule_payment_tution_date")
    private LocalDate schedulePaymentTutionDate;

    @Column(name = "actual_tution_payment_date")
    private LocalDate actualTutionPaymentDate;

    @Column(name = "other_religion")
    private String otherReligion;
    
}

File: backend/app/models/additional_student_info.py
from sqlalchemy import Column, Integer, String, Date, BigInteger, ForeignKey, Boolean
from sqlalchemy.orm import relationship
from app.database import Base

class AdditionalStudentInfo(Base):
    __tablename__ = "additional_student_info"

    id = Column(BigInteger, primary_key=True, index=True)
    common_student_id = Column(BigInteger, ForeignKey("common_students.id"), nullable=False, unique=True)
    
    name_in_japanese = Column(String)
    passport_number = Column(String)
    current_japan_level = Column(String)
    japan_travel_experience = Column(Boolean)
    coe_application_experience = Column(Boolean)
    passed_highest_jlpt_level = Column(String)
    
    secondary_phone = Column(String)
    father_name = Column(String)
    
    desired_job_type = Column(String)
    other_desired_job_type = Column(String)
    is_smoking = Column(Boolean)
    is_alcohol_drink = Column(Boolean)
    have_tatto = Column(Boolean)
    hostel_preference = Column(Boolean)
    memo_notes = Column(String(2000))
    attending_class_related_status = Column(String)
    contact_viber = Column(String)
    schedule_payment_tution_date = Column(Date)
    actual_tution_payment_date = Column(Date)
    other_religion = Column(String)

    common_student = relationship("Student", back_populates="additional_info")
